src/utils/hooks.js

```diff
@@ -19,3 +19,68 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+/**
+ * A custom hook that manages the importance of lists and sorts them accordingly.
+ * @param {Array<Object>} lists - An array of list objects to manage, each containing at least a `path` property.
+ * @returns {Object} An object containing:
+ *  - {Array<Object>} sortedLists - An array of lists sorted by importance and date they were pinned.
+ *  - {Function} setImportantList - A function to toggle the importance of a list identified by its `path`.
+ *  - {Function} isListImportant - A function to check if a given list is marked as important.
+ */
+export function useImportance(lists) {
+	const [sortedLists, setSortedLists] = useState([]);
+	const [importantLists, setImportantLists] = useState(
+		() => JSON.parse(localStorage.getItem('importantLists')) || [],
```

Comment:
If we're reading this correctly, the important lists are saved to local storage only?

Are the important lists the same as pinned?

We were curious to know if we could leverage firebase for this instead...or was the intent for the important lists to be different for each user?